plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergMaterializedView.java

```diff
@@ -126,4 +203,91 @@ AS SELECT sum(value) AS s FROM iceberg2.tpch.common_base_table
         assertUpdate(defaultIceberg, "DROP TABLE common_base_table");
         assertUpdate("DROP MATERIALIZED VIEW mv_on_iceberg2");
     }
+
+    public static class RandomTableFunction
```

Comment:
I don't think I would be able to demonstrate that the fix works with that. 
The problem is that when we create a MV that uses PTF as a source we query that PTF every time MV is queried no matter if it was refreshed or not. 
So with IdentityFunction we would create MV like:
```
CREATE MATERIALIZED VIEW mv  AS SELECT * FROM TABLE(mock.system.identity_function(1))
```
then every time when this mv is queried it would return just 1 so we wouldn't be able to tell if it in fact used data stored in mv or went to the source PTF.

Reply:
Makes sense, thanks.